src/app/_services/workflow.service.ts

```diff
@@ -293,8 +294,9 @@ export class WorkflowService {
   }
 
   // show a prompt to cancel workflow
-  promptCancelThisWorkflow(id: string): void {
-    this.promptCancelWorkflow.emit(id);
+  promptCancelThisWorkflow(datasetId: string, datasetName: string): void {
+    const workflowDesc: WorkflowDesc = { id: datasetId, datasetName } as WorkflowDesc;
+    this.promptCancelWorkflow.emit(workflowDesc);
```

Comment:
you can just use: `this.promptCancelWorkflow.emit({ id: datasetId, datasetName });`. any object that has the right fields, will match `WorkflowDesc`. No need to cast.